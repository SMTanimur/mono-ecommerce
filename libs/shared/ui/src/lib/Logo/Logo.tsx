import React from 'react';

import clsx from 'clsx';
import Link from 'next/link';

import { useIsomorphicLayoutEffect } from '@ecommerce/shared/utils/hooks/useIsomorphicLayoutEffect';

/* eslint-disable-next-line */
export interface LogoProps {}

export const Logo: React.FC<LogoProps> = () => {
    const [firstWord, secondWord] = 'cat shop'.split(' ');

    useIsomorphicLayoutEffect(() => {}, []);

    return (
        <Link href="/">
            <a
                className={clsx('-mb-4 w-max', 'flex flex-col items-center')}
                title="fake shop"
            >
                <div
                    className={clsx(
                        'flex',
                        'font-bold text-2xl leading-4 uppercase',
                        'pointer-events-none select-none',
                        'animate-jello',
                    )}
                    aria-hidden="true"
                    data-testid="first-word"
                >
                    {firstWord.split('').map((char, index) => (
                        <span
                            className={clsx('inline-block', 'animate-bounce')}
                            style={{ animationDelay: `-${index * 50}ms` }}
                            key={char}
                        >
                            {char}
                        </span>
                    ))}
                </div>
                <div
                    className={clsx(
                        'flex',
                        'font-bold text-2xl uppercase',
                        'pointer-events-none select-none',
                        'animate-jello',
                    )}
                    aria-hidden="true"
                    data-testid="second-word"
                >
                    {secondWord.split('').map((char, index) => (
                        <span
                            className={clsx('inline-block', 'animate-bounce')}
                            style={{ animationDelay: `-${index * 50}ms` }}
                            key={char}
                        >
                            {char}
                        </span>
                    ))}
                </div>
            </a>
        </Link>
    );
};
