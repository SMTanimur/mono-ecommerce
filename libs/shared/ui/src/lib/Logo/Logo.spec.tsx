import '@testing-library/jest-dom';
import { RenderResult, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { Logo } from './Logo';

describe('Logo', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
        renderResult = render(<Logo />);
    });

    it('should render successfully', () => {
        const { baseElement } = renderResult;

        const tree = create(<Logo />).toJSON();
        expect(tree).toMatchSnapshot();

        expect(baseElement).toBeTruthy();
    });

    it('contain all chars in two words', () => {
        const [firstWord, secondWord] = 'cat shop'.split(' ');

        const firsWordContainerElement = screen.getByTestId('first-word');

        firstWord.split('').forEach((char) => {
            const charElement = screen.getByText(char);
            expect(charElement).toBeInTheDocument();
            expect(charElement.parentElement).toBe(firsWordContainerElement);
        });

        const secondWordContainerElement = screen.getByTestId('second-word');
        secondWord.split('').forEach((char) => {
            const charElement = screen.getByText(char);
            expect(charElement).toBeInTheDocument();
            expect(charElement.parentElement).toBe(secondWordContainerElement);
        });
    });
});
