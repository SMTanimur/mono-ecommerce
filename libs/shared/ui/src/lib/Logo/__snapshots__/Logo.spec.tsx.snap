// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render successfully 1`] = `
<a
  className="-mb-4 w-max flex flex-col items-center"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  title="fake shop"
>
  <div
    aria-hidden="true"
    className="flex font-bold text-2xl leading-4 uppercase pointer-events-none select-none animate-jello"
    data-testid="first-word"
  >
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-0ms",
        }
      }
    >
      c
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-50ms",
        }
      }
    >
      a
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-100ms",
        }
      }
    >
      t
    </span>
  </div>
  <div
    aria-hidden="true"
    className="flex font-bold text-2xl uppercase pointer-events-none select-none animate-jello"
    data-testid="second-word"
  >
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-0ms",
        }
      }
    >
      s
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-50ms",
        }
      }
    >
      h
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-100ms",
        }
      }
    >
      o
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-150ms",
        }
      }
    >
      p
    </span>
  </div>
</a>
`;
